From e368725c01994f64e189e6280fdb6d0935cc5e6b Mon Sep 17 00:00:00 2001
From: Shiyou Yin <yinshiyou-hf@loongson.cn>
Date: Wed, 29 Jul 2020 18:11:00 +0800
Subject: [PATCH 19/26] avcodec/mips/cabac: Fix a bug in get_cabac_inline_mips.

Failed fate case: fate-h264-conformance-caba2_sony_e
Clang is more strict in the use of register constraint.

Signed-off-by: Michael Niedermayer <michael@niedermayer.cc>
---
 libavcodec/mips/cabac.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavcodec/mips/cabac.h b/libavcodec/mips/cabac.h
index c595915eda..3d09e93523 100644
--- a/libavcodec/mips/cabac.h
+++ b/libavcodec/mips/cabac.h
@@ -109,7 +109,7 @@ static av_always_inline int get_cabac_inline_mips(CABACContext *c,
       [lps_off]"i"(H264_LPS_RANGE_OFFSET),
       [mlps_off]"i"(H264_MLPS_STATE_OFFSET + 128),
       [norm_off]"i"(H264_NORM_SHIFT_OFFSET),
-      [cabac_mask]"i"(CABAC_MASK)
+      [cabac_mask]"r"(CABAC_MASK)
     : "memory"
     );
 
-- 
2.30.1

